### PR TITLE
fix(session): inject GitHub owner/repo into worker prompt from git remote

### LINKED ISSUE #316: fix(session): Worker prompt missing GitHub repo owner/name
## Problem

When a worker is spawned, it doesn't know the correct GitHub repository name. The worker assumes the folder name matches the repo, leading to failed `gh` commands.

## Observed Behavior

1. Spawned worker for issue #276
2. Worker tried: `gh issue view 276 --repo joshsmithxrm/ppds`
3. **Failed**: repo is actually `joshsmithxrm/power-platform-developer-suite`
4. Required manual guidance via `ppds session forward`

## Root Cause

The worker prompt template doesn't extract and inject the GitHub remote URL. Worker guesses based on folder name.

## Proposed Fix

In `SpawnCommand.cs` or worker prompt generation:

```csharp
// Extract from git remote
var remoteUrl = RunGit("remote get-url origin");
var (owner, repo) = ParseGitHubUrl(remoteUrl);
// e.g., "https://github.com/joshsmithxrm/power-platform-developer-suite.git"
//       → owner: "joshsmithxrm", repo: "power-platform-developer-suite"

// Inject into worker prompt header:
"""
## Repository Context
- GitHub: {owner}/{repo}
- Issue: #{issueNumber}
- Branch: issue-{issueNumber}

For GitHub CLI commands, always use:
  gh issue view {issueNumber} --repo {owner}/{repo}
  gh pr create --repo {owner}/{repo} ...
"""
```

## Acceptance Criteria

- [ ] Worker prompt includes correct `owner/repo` from git remote
- [ ] Worker can run `gh issue view` without guidance
- [ ] Works with both HTTPS and SSH remote URLs

### LINKED ISSUE #318: fix(session): Worker uses wrong repo in MCP GitHub tool calls
## Problem

When a worker session uses MCP GitHub tools (like `get_issue`), it uses incorrect repo owner/name:

```
github - get_issue(owner: "Power-Platform-Patterns", repo: "ppds", issue_number: 316)
```

Should be:
```
github - get_issue(owner: "joshsmithxrm", repo: "power-platform-developer-suite", issue_number: 316)
```

## Related

This is the same root cause as #316 (worker prompt missing repo context), but manifests differently:
- #316: Worker uses wrong repo in `gh` CLI commands
- This issue: Worker uses wrong repo in MCP tool calls

## Root Cause

The worker prompt doesn't include GitHub repo information, so the worker:
1. Guesses based on folder name (`ppds`)
2. Or uses some default/cached value (`Power-Platform-Patterns`)

## Proposed Fix

Same fix as #316 - inject correct `owner/repo` from `git remote get-url origin` into the worker prompt. The prompt should explicitly state:

```markdown
## Repository Context
- Owner: joshsmithxrm
- Repo: power-platform-developer-suite

For all GitHub operations (CLI or MCP), use this owner/repo.
```

## Acceptance Criteria

- [ ] Worker prompt includes explicit owner/repo
- [ ] MCP GitHub tool calls use correct repo
- [ ] `gh` CLI commands use correct repo

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

- Extract GitHub owner/repo from `git remote get-url origin` during session spawn
- Add `ParseGitHubUrl` helper method supporting both HTTPS and SSH URL formats
- Inject "Repository Context" section into worker prompt with explicit owner/repo values
- Provide examples for both `gh` CLI commands and MCP GitHub tool calls

## Problem

Workers were guessing the GitHub repository from the folder name, which failed when the folder name differed from the repo name (e.g., folder `ppds` but repo `power-platform-developer-suite`). This caused:
- `gh issue view` commands to fail
- MCP GitHub tool calls to use wrong repo

## Test Plan

- [x] Unit tests for `ParseGitHubUrl` with HTTPS URLs (with/without .git suffix)
- [x] Unit tests for `ParseGitHubUrl` with SSH URLs (with/without .git suffix)
- [x] Unit tests for edge cases (whitespace, invalid URLs)
- [x] All existing tests pass
- [x] Build succeeds with no warnings

Closes #316
Closes #318

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/PPDS.Cli/Services/Session/SessionService.cs [hunk 1]
```diff
@@ -432,6 +436,82 @@ public Task HeartbeatAsync(string sessionId, CancellationToken cancellationToken
...
+        using var process = Process.Start(startInfo)
+            ?? throw new InvalidOperationException("Failed to start git process");
+
+        var output = await process.StandardOutput.ReadToEndAsync(cancellationToken);
+        await process.WaitForExitAsync(cancellationToken);
+
+        if (process.ExitCode != 0)
+        {
+            throw new InvalidOperationException("Failed to get git remote URL");
+        }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/Session/SessionService.cs]: Fixed in 4ae6032 - now reading stderr asynchronously and including detailed error context with repository path and exit code.

File: src/PPDS.Cli/Services/Session/SessionService.cs [hunk 2]
```diff
@@ -432,6 +436,82 @@ public Task HeartbeatAsync(string sessionId, CancellationToken cancellationToken
...
+
+        return output.Trim();
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/Session/SessionService.cs]: Fixed in 4ae6032 - applied the suggested improvement with descriptive error message including repo path, exit code, and git stderr output.

[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/Session/SessionService.cs]: Fixed in 4ae6032 - now reading both stdout and stderr asynchronously to avoid deadlocks, and including stderr in error message for better diagnostics.

File: src/PPDS.Cli/Services/Session/SessionService.cs [hunk 3]
```diff
@@ -432,6 +436,82 @@ public Task HeartbeatAsync(string sessionId, CancellationToken cancellationToken
...
+        // SSH format: git@github.com:owner/repo
+        if (url.StartsWith("git@github.com:", StringComparison.OrdinalIgnoreCase))
+        {
+            var path = url["git@github.com:".Length..];
+            var parts = path.Split('/', StringSplitOptions.RemoveEmptyEntries);
+            if (parts.Length >= 2)
+            {
+                return (parts[0], parts[1]);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/Session/SessionService.cs:519]: Fixed in 4ae6032 - refactored to extract common path-splitting logic, eliminating the code duplication.
